src/lib/cache/SubscriptionCache.cpp

```diff
@@ -0,0 +1,1000 @@
+
+  //
+  // 0. Fill in 'constant initial values'
+  //
+  lastNotificationTime  = -1;
+  pendingNotifications  = 0;
+  format                = XML;
+
+
+
+  //
+  // 1. Get Tenant, ServicePath and Subscription Id
+  //
+  tenant         = _tenant;
+  servicePath    = _servicePath;
+  subscriptionId = _subscriptionId;
+  format         = _format;
+
+  //
+  // 2. Convert all patterned EntityId to EntityInfo
+  //
+  for (ix = 0; ix < scrP->entityIdVector.vec.size(); ++ix)
+  {
+    EntityId* eIdP;
+
+    eIdP = scrP->entityIdVector.vec[ix];
+
+    if (eIdP->isPattern == "false")
+    {
+      continue;
+    }
+
+    LM_M(("KZ: Subscription: adding EntityInfo with pattern '%s'", eIdP->id.c_str()));
+    EntityInfo* eP = new EntityInfo(eIdP->id, eIdP->type);
+    
+    entityIdInfoAdd(eP);
+  }
+
+
+  //
+  // 3. Insert all attributes 
+  //
+  for (ix = 0; ix < scrP->attributeList.attributeV.size(); ++ix)
+  {
+    attributes.push_back(scrP->attributeList.attributeV[ix]);
+  }
+
+
+  //
+  // 4. throttling
+  //
+  throttling = scrP->throttling.parse();
+
+
+  //
+  // 5. expirationTime
+  //
+  expirationTime = _expiration;
+
+
+  //
+  // 6. restriction
+  //
+  restriction.fill(&scrP->restriction);
+
+
+  //
+  // 7. notifyConditionVector
+  //
+  notifyConditionVector.fill(scrP->notifyConditionVector);
+  LM_M(("Added %d notifyConditions", notifyConditionVector.size()));
+
+  //
+  // 8. reference
+  //
+  reference.set(scrP->reference.get());
+}
+
+
+
+/* ****************************************************************************
+*
+* Subscription::Subscription - 
+*/
+Subscription::Subscription
+(
+  const std::string&               _tenant,
+  const std::string&               _servicePath,
+  const std::string&               _subscriptionId,
+  const std::vector<EntityInfo*>&  _entityIdInfos,
+  const std::vector<std::string>&  _attributes,
+  int64_t                          _throttling,
+  int64_t                          _expirationTime,
+  const Restriction&               _restriction,
+  NotifyConditionVector&           _notifyConditionVector,
+  const std::string&               _reference,
+  Format                           _format
+)
+{
+  unsigned int ix;
+
+  tenant         = _tenant;
+  servicePath    = _servicePath;
+  subscriptionId = _subscriptionId;
+  format         = _format;
+
+  for (ix = 0; ix < _entityIdInfos.size(); ++ix)
+  {
+    entityIdInfos.push_back(_entityIdInfos[ix]);
+  }
+
+  for (ix = 0; ix < _attributes.size(); ++ix)
+  {
+    attributes.push_back(_attributes[ix]);
+  }
+
+  throttling     = _throttling;
+  expirationTime = _expirationTime;
+
+  restriction.fill((Restriction*) &_restriction);
+  notifyConditionVector.fill(_notifyConditionVector);
+  reference.set(_reference);
+}
+
+
+
+/* ****************************************************************************
+*
+* Subscription::entityIdInfoAdd - 
+*
+* FIXME P10: need to decide whether this function copies the EntityInfo or
```

Comment:
I simply didn't know whether it would work correctly just copying the pointer, later the tests seem to verify that it is enough. Simply removing the FIXME.